torch/onnx.py

```diff
@@ -23,35 +44,60 @@ def export(model, args, f, export_params=True, kwargs=None, verbose=False):
 
     Arguments:
         model (torch.nn.Module): the model to be exported.
-        args (torch.autograd.Variable or tuple of variables): the inputs to
-            the model, e.g., such that ``model(*args, **kwargs)`` is a valid
-            invocation of the model (see kwargs below).
+        args (tuple of arguments): the inputs to
+            the model, e.g., such that ``model(*args)`` is a valid
+            invocation of the model.  Any non-Variable arguments will
+            be hard-coded into the exported model; any Variable arguments
+            will become inputs of the exported model, in the order they
+            occur in args.  If args is a Variable, this is equivalent
+            to having called it with a 1-ary tuple of that Variable.
+            (Note: passing keyword arguments to the model is not currently
+            supported.  Give us a shout if you need it.)
         f: a file-like object (has to implement fileno that returns a file descriptor)
             or a string containing a file name.  A binary Protobuf will be written
             to this file.
         export_params (bool, default True): if specified, all parameters will
-            be exported.  Set this to False if you are exporting an
-            untrained model.
-        kwargs (dict, optional): keyword inputs to the model.
+            be exported.  Set this to False if you want to export an untrained model.
+            In this case, the exported model will first take all of its parameters
+            as arguments, the ordering as specified by ``model.state_dict().values()``
+        verbose (bool, default False): if specified, we will print out a debug
+            description of the trace being exported.
+        training (bool, default False): export the model in training mode.  At
+            the moment, ONNX is oriented towards exporting models for inference
+            only, so you will generally not need to set this to True.
     """
-    _export(model, args, f, export_params, kwargs, verbose)
+    _export(model, args, f, export_params, verbose, training)
 
 
-def _export(model, args, f, export_params=True, kwargs=None, verbose=False):
+def _export(model, args, f, export_params=True, verbose=False, training=False):
```

Comment:
why did you remove `kwargs`?

Reply:
They didn't actually do the right thing. The docs implied that they would be passed to the model as keyword arguments, but they didn't actually do that.